src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -281,11 +287,14 @@ HashSet<ITypeSymbol> typeStack
 				}
 
 				// We descend into the base class last
-				var baseResult = InspectBaseType( type, typeStack );
-				if( baseResult.IsMutable ) {
-					return baseResult;
+				if ( type.BaseType != null ) {
+					var baseResult = InspectConcreteType( type.BaseType, typeStack );

```

Comment:
`type.BaseType` could still be `abstract`, I believe, but I don't think there's any functional issue if it was. I don't know if there's a more appropriate name for the function though.

Reply:
That's true. "Concrete type" is a D2L-ism that kinda bugs me but I haven't found a more suitable term looking at the literature.

Reply:
If you can think of a better name I'd be happy to switch stuff. Things I've considered (none:

* BaseType (already has a meaning in C# that doesn't precisely match)
* BaseClass (doesn't really make sense generally e.g. structs )
* SpecificType (name is... unspecific, no better than concrete type IMO)